packages/fluentui/react-northstar/test/specs/commonTests/utils/getComponentPath.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { mount } from 'enzyme';
+import config from '@uifabric/build/config';
+
+// hack to work around lack of esModuleInterop
+import * as StackUtilsType from 'stack-utils'; // eslint-disable-line
+const StackUtils: typeof StackUtilsType = require('stack-utils');
+
+// Stack cleaner that should remove stack entries *except* the ones from the component itself
+const stackUtils = new StackUtils({
+  cwd: config.paths.base(),
+  internals: StackUtils.nodeInternals().concat(/node_modules/, /react[\\/]test/, /anonymous/, /\butils\b/),
+});
+const reactComponentsRegex = /[\\/]react[/\\]src[/\\]components[\\/]/;
+
+/**
+ * Get the file path within the repo for a component.
+ */
+export default function getComponentPath(Component: React.ComponentType, constructorName: string): string {
+  // Prevent framework code from logging errors
+  const consoleErrorMock = jest.spyOn(console, 'error').mockImplementation();
+  const consoleWarnMock = jest.spyOn(console, 'warn').mockImplementation();
+  // Hack to make UIComponent constructor throw an exception
+  (window as any).gettingComponentPath = true;
+
+  let error = '';
+  let result = '';
+
+  try {
+    // Attempt to cause an exception to be thrown from within the component code, so we can get
+    // the path from the stack.
+    if (Component.prototype) {
+      // Class component. Try a couple different approaches.
+      new (Component as React.ComponentClass)(null); // eslint-disable-line
+      mount(<Component />);
+    } else {
+      // Function component.
+      // Calling as a function (not rendering) will likely result in an invalid hook error.
```

Comment:
And if not?

Reply:
In that case the test will fail and we'll find another way to make it work with the newly tested component.